apps/sms/views/shared/js/settings.js

```diff
@@ -67,9 +70,37 @@ var Settings = {
       }
     }
 
+    //create lock for SEND_READ_REPORT_KEY
+    keyHandlerSet[this.SEND_READ_REPORT_KEY] = this.toSendReadReport.bind(this);
+
     for (var key in keyHandlerSet) {
       setHandlerMap(key);
     }
+
+    //add Observer for SEND_READ_REPORT_KEY
+    settings.addObserver(this.SEND_READ_REPORT_KEY, function (event) {
+      Settings.sendReadReportStatus = event.settingValue;
+      console.log(Settings.sendReadReportStatus);
+    });
+  },
+
+  sendReadReport: function toSendReadReport() {
+    var deferred = Utils.Promise.defer();
+    if(Settings.sendReadReportStatus != null) {
+      deferred.resolve(Settings.sendReadReportStatus);
+    } else {
+      deferred.reject('ERROR: Settings.sendReadReportStatus unavailable');
+    };
+
+    return deferred.promise;
+  },
+
+  //Set sendReadReportStatus for SEND_READ_REPORT_KEY
+  toSendReadReport: function toSendReadReport(status) {
+    if (!isNaN(status)) {
+      Settings.sendReadReportStatus = status;
+      console.log(Settings.sendReadReportStatus);
+    }
   },
```

Comment:
@julienw i am just curious to know why we have  this._sendReadReportDefer = Utils.Promise.defer(); in init and resolving it in initSendReadReport.
Instead we can have this._sendReadReportDefer=Promise.resolve(value) directly in initSendReadReport.
I guess reason is , setHandlerMap(key) function in init has aysnc api (DOMrequest).
please correct it if i am wrong.

Also, few days back , steve warned me that, we shouldn't go for sync looping like foreach,while for looping having async method inside. That's why we go for chaining of the api call.
Similarly  here too, 
  <pre>
  for (var key in keyHandlerSet) {
      setHandlerMap(key); //this method inisde it has async DOM request.
    }
</pre>